### PR TITLE
[Feature] Add id field to Catalog

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

use catalog_id as reserved fields for later support operations such as rename

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 5 / 7 (71.43%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/InternalCatalog.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/catalog/Catalog.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/catalog/ExternalCatalog.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java [hunk 1]
```diff
@@ -68,7 +68,8 @@ public synchronized void createCatalog(CreateCatalogStmt stmt) throws DdlExcepti
         }
 
         connectorMgr.createConnector(new ConnectorContext(catalogName, type, properties));
-        Catalog catalog = new ExternalCatalog(catalogName, comment, properties);
+        long id = GlobalStateMgr.getCurrentState().getNextId();
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java:71]: should I add `idToCatalog` in this pr?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 23 / 24 (95.83%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/InternalCatalog.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|19|19|100.00%|
|:large_blue_circle:|com/starrocks/catalog/ExternalCatalog.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/Catalog.java|3|3|100.00%|

**Review by caneGuy — COMMENTED** [review]
code LGTM, a question about this pr. what is the usage of this field?

**Comment by stephen-shelby** [comment]
> 
id is reserved for future use

**Comment by stephen-shelby** [comment]
> code LGTM, a question about this pr. what is the usage of this field?

id is reserved for future use

**Review by caneGuy — APPROVED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 5 / 7 (71.43%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/InternalCatalog.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/catalog/Catalog.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/catalog/ExternalCatalog.java|1|1|100.00%|

**Review by gengjun-git — APPROVED** [review]

**Comment by stephen-shelby** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**